### DISCUSSION TITLE
Supabase urgently needs per project permissions

### DISCUSSION THREAD

**Discussion by bombillazo** [supabase/supabase#20942] [Feature Requests]
Hey, we are having an unforeseen bad development and operational experience due to how permissions are global across projects in an organization. We cannot (and **_should_** not) give access to developers, agents, and operations team to our production environment. Still, they require access to other testing, staging, and administrative projects we run on Supabase. We are having a lousy time juggling permissions and limiting access to the projects.

This is not an uncommon scenario. I also imagine having an organization serving multiple clients with different projects but having trouble inviting client users to the dashboard since that opens up the projects of other clients.

Is there any expectation that this behavior will change? It will significantly improve our experience of using the Supabase platform.

**Comment by etbyrd** [comment]
+1 This is a extremely annoying problem. Its made worse considering that the functionality exists - but only for enterprise customers.

**Comment by hugopqt** [comment]
I completely agree. Who thought it was a good idea? We need permissions per project, not per organisations.
